### PR TITLE
Remove inline js for gantt chart

### PR DESCRIPTION
**Pull request by bbovenzi**
Continuation of #14115.
- Removing inline js for the Gantt chart. 
- Linting all Gantt js
- Use more conventional file name

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by ryanahamilton — APPROVED** [review]
Nice work @bbovenzi!

**Review by ryanahamilton — CHANGES_REQUESTED** [review]
Appears to be some related test failures:
```bash
 =================================== FAILURES ===================================
  _________________ TestExtraLinks.test_extra_link_in_gantt_view _________________
  
  self = <tests.www.test_views.TestExtraLinks testMethod=test_extra_link_in_gantt_view>
  
      def test_extra_link_in_gantt_view(self):
          exec_date = dates.days_ago(2)
          start_date = datetime(2020, 4, 10, 2, 0, 0)
          end_date = exec_date + timedelta(seconds=30)
      
          with create_session() as session:
              for task in self.dag.tasks:
                  ti = TaskInstance(task=task, execution_date=exec_date, state="success")
                  ti.start_date = start_date
                  ti.end_date = end_date
                  session.add(ti)
      
          url = f'gantt?dag_id={self.dag.dag_id}&execution_date={exec_date}'
          resp = self.client.get(url, follow_redirects=True)
      
  >       self.check_content_in_response('"extraLinks":', resp)
  
  tests/www/test_views.py:2940: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = <tests.www.test_views.TestExtraLinks testMethod=test_extra_link_in_gantt_view>
  text = '"extraLinks":', resp = <Response 46988 bytes [200 OK]>, resp_code = 200
  
      def check_content_in_response(self, text, resp, resp_code=200):
          resp_html = resp.data.decode('utf-8')
          assert resp_code == resp.status_code
          if isinstance(text, list):
              for line in text:
                  assert line in resp_html
          else:
  >           assert text in resp_html
  E           AssertionError: assert '"extraLinks":' in "\n\n\n\n\n\n\n\n\n\n    \n\n\n<!DOCTYPE html>\n<html>\n  <head>\n    <title>dag - Gantt - Airflow</title>\n\n    \n  ...   $input.addClass('switch-input--error');\n        }, 500 );\n      });\n    });\n  </script>\n\n\n  </body>\n</html>"
 
```

File: airflow/www/templates/airflow/gantt.html [hunk 1]
```diff
@@ -76,20 +76,11 @@
...
-        .height(data.height)
-        .selector('.gantt')
-        .tickFormat("%H:%M:%S");
-      gantt(data.tasks);
-      $('body').on('airflow.timezone-change', () => {
-        gantt.redraw(data.tasks);
-      });
-    });
+    // Loading data via <meta> wasn't loading extra_links in a task
+    const data = JSON.parse('{{ data|tojson }}');
```
Comment (bbovenzi) [inline airflow/www/templates/airflow/gantt.html:82]: @ryanahamilton Let me know if this is ok.

**Comment by ryanahamilton** [comment]
@bbovenzi looks like a new conflict to resolve. LMK once you resolve that so we can merge this ASAP (and avoid additional ones).

**Comment by bbovenzi** [comment]
Something weird happened with some rebase going to start a fresh pr.

**Comment by ryanahamilton** [comment]
Replaced by #14545.

**Pull request closed without merging**